src/CoreBluetooth/AdvertisementData.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Linq;
+
+using XamCore.ObjCRuntime;
+using XamCore.Foundation;
+using XamCore.CoreFoundation;
+
+namespace XamCore.CoreBluetooth {
+
+	public partial class AdvertisementData {
+		
+		public IDictionary <CBUUID, NSData> ServiceData {
+			get {
+				using (var nsDict = _ServiceData) {
+					var result = new Dictionary <CBUUID, NSData> ();
+					foreach (var k in nsDict.Keys) {
+						var nsData = nsDict.ObjectForKey (k) as NSData;
+						var uuid = k as CBUUID;
+						if (uuid != null && nsData != null)
+							result.Add (uuid, nsData);
+					}
+					return result;
+				}
+			}
+			set {
+				_ServiceData = NSDictionary.FromObjectsAndKeys (value.Values.ToArray (), value.Keys.ToArray ());
```

Comment:
`a.ServiceData = a.ServiceData;` becomes most costly (with both managed arrays).

@dalexsoto you made more strong dictionaries that most people, are there many cases where we're not using references ? (but copies)